### ISSUE TITLE
Ideas list

### ISSUE THREAD

**Issue by kandersolar** [AssessingSolar/unofficial-psm3-userguide#1]
- [ ] Masking out non-land pixels
    + https://regionmask.readthedocs.io/en/stable/defined_landmask.html
    + https://stackoverflow.com/a/48062502/1641381
- [x] Pixel boundaries for each dataset
    + 5-min: http://kevbase.com/site/posts/psm3-pixels-boundaries.html
- [ ] Pixel area and shape
- [x] Interactive pixel map like [PRISM](https://www.prism.oregonstate.edu/explorer/)
- [ ] Replicate auxiliary data (temperature, etc):
    + > Water vapor is obtained from NASA MERRA 2. Additional meteorological parameters are also derived from MERRA 2.
    + Wind speed and direction show an odd artifact; see the "pulses" in their traces in this plot: https://github.com/AssessingSolar/unofficial-psm3-userguide/pull/7#issuecomment-1401308090
    + Temperature is downscaled from ~50 km MERRA 2 data using a simple lapse rate:
        * 6.5 degrees per km elevation: https://www.nrel.gov/docs/fy21osti/77635.pdf
        * 6.56 degrees per km: https://nrel.github.io/rex/_autosummary/rex.renewable_resource.NSRDB.html#rex.renewable_resource.NSRDB.LAPSE_RATES
        * 6 degrees per km: https://doi.org/10.1016/j.rser.2018.03.003
- [ ] Replicate the irradiance calculations?
    + > PSM uses a two-step process where cloud properties are retrieved using the adapted PATMOS-X model, which are then used as inputs to REST2 for clear sky and FARMS for cloudy sky radiation calculations. REST2 calculates both DNI and GHI. FARMS calculates GHI, and the DISC model is then used to calculate DNI. Aerosol properties are estimated using MODIS, MISR, and AERONET products. 
- [x] Recreate solar position values using pvlib (especially what specific coordinates are used -- center of pixel, requested location, etc)
- [x] Show pvlib's DISC reproduces cloudy-sky DHI
- [x] Replicate lower-res values from 5-minute
- [x] Rate limit info
- ~[ ] Concurrent requests info~ concurrent requests no longer seem to be rejected, so nevermind about this
- [x] TMY does not perfectly match underlying single-years
- [x] End-of-year UTC-offset "roll-over" effect
- [x] Auxiliary HSDS data (land cover, population, etc)
- [x] The 23:00 UTC Tamb artifact in 5-minute data
- [ ] Interval timestamp labeling conventions
- [ ] Is there any difference between the two endpoints that supply 30/60-minute data?
- [x] Data precision -- number of decimals, rounding, etc
- [ ] What does "clear-sky" mean to PSM3 wrt cloud type and the irradiance data?
- [ ] List of other information sources like the webinar Q&A docs? 
    + https://nsrdb.nrel.gov/nsrdb/NSRDB_Webinar_QA_-_Oct_6_2020.pdf
    + https://www.youtube.com/watch?v=BzNoYACXCOY
- [ ] Are irradiance values different for pixels over water?
- [ ] Variables names (including differences between API request and response)
- [ ] Recreate UV irradiance doses
- [x] Recreate albedo values (https://doi.org/10.1029/JC092iC12p13115)

**Comment by AdamRJensen** [comment]
> * [ ]  Illustrations that 5-minute PSM3 data does not contain the same variability as 5-min average irradiance ground weather station measurements or falling on a single PV inverter (this has been done elsewhere, but would be good to have in an NSRDB user guide so the data are not misused)

@williamhobbs df['ghi'].diff().plot.hist() do the trick? I.e., a histogram of the difference between the individual time stamps. Prehaps also a normal histogram of the ghi/dni/dhi values?

**Comment by williamhobbs** [comment]
Yes, I think a histogram of the difference would work. And a histogram/PDF could be informative. 

Something based on pvanalytics.metrics.variability_index could be useful, as well as CDFs of GHI etc. values. 

We used a CDF of GHI values here https://doi.org/10.1109/PVSC48317.2022.9938632 (preprint: https://www.nrel.gov/docs/fy22osti/82812.pdf) to show that is doesn't contain the same range of values.

![image](https://user-images.githubusercontent.com/45701090/205938116-49d420ec-f666-48f1-8da6-203c9ed39c8b.png)

**Comment by williamhobbs** [comment]
Thinking about this more, the range of values overall (like shown in the CDF in my last comment) and the range of values within hours relative to average for the hour are what matters for subhourly clipping loss error. I often incorrectly equate those with "variability" - they are similar and often correlated, but not the same.
